plugins/ccgx/fu-ccgx-i2c.c

```diff
@@ -0,0 +1,774 @@
+	CY_GPIO_SET_CONFIG_CMD,		/* set the GPIO configuration */
+	CY_GPIO_GET_VALUE_CMD,		/* get GPIO value */
+	CY_GPIO_SET_VALUE_CMD,		/* set the GPIO value */
+	CY_PROG_USER_FLASH_CMD = 0xE0,	/* program user flash area. The total space available is 512 bytes
+					 * this can be accessed by the user from USB. The flash	area
+					 * address offset is from 0x0000 to 0x00200 and can be written to
+					 * page wise (128 byte) */
+	CY_READ_USER_FLASH_CMD,		/* read user flash area. The total space available is 512 bytes
+					 * this	can be accessed by the user from USB. The flash	area
+					 * address offset is from 0x0000 to 0x00200 and can be written to
+					 * page wise (128 byte) */
+	CY_DEVICE_RESET_CMD = 0xE3,	/* performs a device reset from firmware */
+} CyVendorCommand;
+
+typedef struct __attribute__((packed)) {
+	guint32	 frequency;		/* frequency of operation. Only valid values are 100KHz and 400KHz */
+	guint8	 slave_address;		/* slave address to be used when in slave mode */
+	guint8	 is_msb_first;		/* whether to transmit most significant bit first */
+	guint8	 is_master;		/* whether to block is to be configured as a master*/
+	guint8	 s_ignore;		/* ignore general call in slave mode */
+	guint8	 is_clock_stretch;	/* wheteher to stretch clock in case of no FIFO	availability */
+	guint8	 is_loop_back;		/* whether to loop back	TX data to RX. Valid only for debug purposes */
+	guint8	reserved[6];		/* reserved for future use */
+} CyUsI2cConfig_t;
+
+static gboolean
+i2c_reset (FuDevice *device, CyI2CDeviceHandle *handle, guint8 mode, GError **error)
+{
+	GUsbDevice *usb_device = NULL;
+	guint32 io_timeout = FU_CCGX_I2C_WAIT_TIMEOUT;
+	guint16 scb_index = 0;
+	guint16 w_value, w_index, w_length;
+	guint8 bm_request;
+	gsize actual_length = 0;
+	g_autoptr(GError) error_local = NULL;
+
+	g_return_val_if_fail (handle != NULL, FALSE);
+	g_return_val_if_fail (FU_USB_DEVICE (device), FALSE);
+	usb_device = fu_usb_device_get_dev (FU_USB_DEVICE (device));
+	g_return_val_if_fail (usb_device != NULL, FALSE);
+
+	scb_index = handle->inf_num;
+	if (scb_index > 0)
+		scb_index = 1;
+
+	bm_request = CY_I2C_RESET_CMD;
+	w_value = ((scb_index << CY_SCB_INDEX_POS) | mode);
+	w_index = 0;
+	w_length = 0;
+
+	if (!g_usb_device_control_transfer (usb_device,
+					   G_USB_DEVICE_DIRECTION_HOST_TO_DEVICE,
+					   G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					   G_USB_DEVICE_RECIPIENT_DEVICE,
+					   bm_request,
+					   w_value,
+					   w_index,
+					   NULL,
+					   w_length,
+					   &actual_length,
+					   io_timeout,
+					   NULL,
+					   &error_local)) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "failed to reset i2c:%s", 
+			     error_local->message);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+i2c_get_status (FuDevice *device, CyI2CDeviceHandle* handle, guint8 mode, guint8 *i2c_status, GError **error)
+{
+	g_autoptr(GError) error_local =	NULL;
```

Comment:
g_autoptr should be at end of the declarations

Reply:
I will move it to end of the declarations.

Reply:
I fixed it.